libsrc/c64/soft80_conio.s

```diff
@@ -51,13 +51,9 @@ soft80_init:
         jmp     soft80_kclrscr
 
 soft80_shutdown:
-        lda     #$1b
-        sta     VIC_CTRL1
-        lda     #$03
-        sta     CIA2_PRA
-        lda     #$15
-        sta     VIC_VIDEO_ADR
-        rts
+
+        jsr     $fda3   ; Initialise I/O
+        jmp     $ff5b   ; Initialize screen editor
```

Comment:
That first subroutine call cannot be used because it re-enables the BASIC ROM.  If a large program (such as Contiki's `webbrowser80`) has destructor code or data "behind" that ROM, then the program might crash when it tries to quit gracefully.  Changing that code to the following works well enough:

~~~gas
        lda     #$07
        sta     CIA2_PRA
        jmp     $FF5B           ; Initialize video I/O
~~~